pkg/controller/storagecluster/external_resources.go

```diff
@@ -171,6 +172,28 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterResources(scs []*s
 		}
 		objectKey := types.NamespacedName{Name: d.Name, Namespace: instance.Namespace}
 		switch d.Kind {
+		case "CephCluster":
+			monitoringIP, ok := d.Data["MonitoringEndpoint"]
+			if !ok {
+				err := fmt.Errorf(
+					"Monitoring Endpoint not present in the external cluster secret %s",
+					externalClusterDetailsSecret)
+				reqLogger.Error(err, err.Error())
+				return err
+			}
+			monitoringPort, ok := d.Data["MonitoringPort"]
+			if !ok {
+				err := fmt.Errorf(
+					"Monitoring Port not present in the external cluster secret %s",
+					externalClusterDetailsSecret)
+				reqLogger.Error(err, err.Error())
+				return err
+			}
+			err := validateMonitoringEndpoint(monitoringIP, monitoringPort, reqLogger)
+			if err != nil {
+				reqLogger.Error(err, "Monitoring validation failed.")
+				return err
+			}
```

Comment:
If you could combine the functionalities into a single one as @umangachapagain has suggested that will be much appreciated...

```suggestion
		case "CephCluster":
			monitoringIP, okMonIP := d.Data["MonitoringEndpoint"]
			monitoringPort, okMonPort := d.Data["MonitoringPort"]
			// if any of the above is missing return an error
			if !okMonIP || !okMonPort {
				err := fmt.Errorf(
					"Both 'MonitoringEndpoint' and 'MonitoringPort' values should be present in secret %s",
					externalClusterDetailsSecret)
				reqLogger.Error(err, err.Error())
				return err
			}
			err := validateMonitoringEndpoint(monitoringIP, monitoringPort, reqLogger)
			if err != nil {
				reqLogger.Error(err, "Monitoring validation failed.")
				return err
			}
```

Reply:
I will refactor the code. Thanks.